### PR TITLE
codegen+runtime: Add array concatenation via `+` and `+=` operators

### PR DESCRIPTION
**Pull request by robryanx**
Resolves: #1148

### CODE REVIEW

File: runtime/Builtins/Array.h [hunk 1]
```diff
@@ -224,6 +224,11 @@ class Array {
         return array;
     }
 
+    ErrorOr<void> operator+=(Array<T> const& b)
```
Comment (ADKaster) [inline runtime/Builtins/Array.h:227]: I'm not sure about this one. The canonical binary operators are supposed to return T or T&, not void.

https://en.cppreference.com/w/cpp/language/operators

This should probably return *this; via `ErrorOr<Array<T&>>`

I'm surprised it's valid to return ErrorOr from these, it looks quite strange. But if it compiles and works then 🤷‍♂️

Comment (robryanx) [reply runtime/Builtins/Array.h:227]: Can't seem to make that one work:
```cpp
ErrorOr<Array<T>&> operator+=(Array<T> const& b)
{
    TRY(push_values(b));
    return *this;
}
```
Gives me:
```
conversion from 'JaktInternal::Array<Jakt::String>' to 'ErrorOr<Array<Jakt::String> &>' is ambiguous
```

Having a look in serenity there are no cases where a += operator uses ErrorOr, there is ByteBuffer though which has
```cpp
void operator+=(ByteBuffer const& other)
{
    MUST(try_append(other.data(), other.size()));
}
```

**Comment by AtkinsSJ** [comment]
This has conflicts.

**Comment by robryanx** [comment]
> This has conflicts.

Yeah, will go through and rebase my prs this weekend.

File: runtime/Builtins/Array.h (hunk 1 above)
Comment (trflynn89) [reply runtime/Builtins/Array.h]: Just happened to be perusing issues this morning - FYI we can't store a reference in an `ErrorOr`. The underlying storage is a `Variant`, which itself can't hold references. I'm surprised this didn't hit an error on this constraint:
https://github.com/SerenityOS/serenity/blob/5b4818df22769bc3353c33b1f1a8d299d96e6efc/AK/Variant.h#L225

Comment (robryanx) [reply runtime/Builtins/Array.h]: Any better way that you know of or is how the PR currently has it best?

Comment (trflynn89) [reply runtime/Builtins/Array.h]: I'm not familiar with the context, but I would suggest avoiding operator overloading entirely, and use a fallible named method instead.

**Pull request closed without merging**